### PR TITLE
A number of updates to the plotting library

### PR DESCRIPTION
**Pull request by DriesAllaerts**
Updates and extensions include:
- Option to specify custom field labels
- Option to specify figure and axes handles
- Option to hide colorbar in `plot_timeheight`
- Option to set dataset-specific plotting options like linestyles and markers
- fig and ax are returned were it like `plt.subplots()`
- Default plotting style of `plot_timehistory_at_height` is set to lines and no markers
- `plot_timehistory_at_height` accepts dataFrame with a single height
- `plot_timehistory_at_height` avoids interpolation when all height(s) exists in dataFrame
- Title fontsize is consistently set to 16
- FutureWarning concerning registration of matplotlib converter is removed

### CODE REVIEW

**Comment by ewquon** [comment]
👏

**Comment by ewquon** [comment]
Thanks for taking the time to clean this up even more, the example looks great to me.

File: plotting.py [hunk 1]
```diff
@@ -309,6 +299,12 @@ def plot_timehistory_at_height(datasets,
         field or height specific colors, limits, etc.
         Example uses include setting linestyle/width, marker, etc.
     """
+    # Avoid FutureWarning concerning the use of an implicitly registered
```
Comment (ewquon) [inline plotting.py]: Ugh, this is ugly, but I've seen it in other code as well... Is there a reference for where this "fix" comes from?

Comment (DriesAllaerts) [reply plotting.py]: I agree, but leaving a big red warning block inside an example notebook seemed even worse. I basically just took the solution suggested in the FutureWarning message

**Pull request merged**